src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
i think traits are more suitable than base classes for this.

regareing body: these would be value objects, no factory? i kind of wonder
if its not the same problem as the request factory has to solve. maybe drop
the methods in favor of the factory?